src/mrsimulator/spin_system/isotope.py

```diff
@@ -21,6 +24,50 @@
 }
 
 
+class IsotopeReferences:
+    _instance = None
+
+    def __new__(cls, *args, **kwargs):
+        if not cls._instance:
+            cls._instance = super().__new__(cls, *args, **kwargs)
+        return cls._instance
+
+    def __init__(self):
+        if not hasattr(self, "initialized"):  # Ensure __init__ is only called once
+            self.initialized = True
+            self.isotopes_data = REFERENCE_DATA
+
+    def get_isotope_data(self, symbol: str):
+        return self.isotopes_data.get(symbol, None)
+
+
+class IsotopeReference:
+    def __init__(self, symbol, data):
+        self.ratio = data.get("ratio") / 100
+        self.compound = data.get("compound")
+        self.solvent = data.get("solvent")
+        self.concentration = data.get("concentration")
+
+    def __repr__(self):
+        return json.dumps(
+            {
+                "ratio": self.ratio,
+                "compound": self.compound,
+                "solvent": self.solvent,
+                "concentration": self.concentration,
+            },
+            indent=4,
+        )
```

Comment:
This section of the code is inherited form pydantic BaseModel class. No need to write it again.

Reply:
Done